Dockerfile-go-deps

```diff
@@ -16,3 +18,79 @@ RUN bin/dep version
 COPY Gopkg.toml Gopkg.lock ./
 RUN bin/dep ensure -vendor-only -v
 
+# Move the dependencies to where `go install` can see them.
+ENV GOPATH=/go
+RUN mv vendor/* $GOPATH/src/
+
+# Precompile key slow-to-build dependencies. This list doesn't need to be
+# complete for the build to work correctly; the completeness of this list
+# only affects the speed of incremental rebuilds of Dependent Dockerfiles.
+#
+# This list was derived from the output of `find $GOPATH/pkg -type f`
+# after building the controller.
```

Comment:
> is it possible to derive this list from the contents of $GOPATH/src/?

I don't think so, for the same reason `go install -v ./vendor/...` doesn't work: there are packages that we have source code for that we don't actually want to build, and that we can't build because `dep` doesn't download dependencies of unneeded (sub-)packages.

I agree this is ugly but I suggest we just go with it for now until we find a better solution. AFAICT a proper solution requires parsing `Gopkg.lock`, which is too much work to do here, especially if we thing Go 1.10 will soon make this all obsolete via #341.